.buildkite/scripts/health-report-tests/bootstrap.py

```diff
@@ -0,0 +1,99 @@
+import util
+import yaml
+
+
+class Bootstrap:
+    ELASTIC_STACK_VERSIONS_URL = "https://artifacts-api.elastic.co/v1/versions"
+
+    def __init__(self) -> None:
+        f"""
+        A constructor of the {Bootstrap}.
+        Returns:
+            Resolves Logstash branch considering provided LS_VERSION
+            Checks out git branch
+        """
+        logstash_version = os.environ.get("LS_VERSION")
+        if logstash_version is None:
+            # version is not specified, use the main branch, no need to git checkout
+            print(f"LS_VERSION is not specified, using main branch.")
+        else:
+            # LS_VERSION accepts major latest as a major.x or specific version as X.Y
+            if logstash_version.find(".x") == -1:
+                print(f"Using specified branch: {logstash_version}")
+                util.git_check_out_branch(logstash_version)
+            else:
+                major_version = logstash_version.split(".")[0]
+                if major_version and major_version.isnumeric():
+                    resolved_version = self.__resolve_latest_stack_version_for(major_version)
+                    minor_version = resolved_version.split(".")[1]
+                    branch = major_version + "." + minor_version
+                    print(f"Using resolved branch: {branch}")
+                    util.git_check_out_branch(branch)
+                else:
+                    raise ValueError(f"Invalid value set to LS_VERSION. Please set it properly (ex: 8.x or 9.0) and "
+                                     f"rerun again")
+
+    def __resolve_latest_stack_version_for(self, major_version: str) -> None:
+        resolved_version = ""
+        response = util.call_url_with_retry(self.ELASTIC_STACK_VERSIONS_URL)
+        release_versions = response.json()["versions"]
+        for release_version in reversed(release_versions):
+            if release_version.find("SNAPSHOT") > 0:
+                continue
+            if release_version.split(".")[0] == major_version:
+                print(f"Resolved latest version for {major_version} is {release_version}.")
+                resolved_version = release_version
+                break
+
+        if resolved_version == "":
+            raise ValueError(f"Cannot resolve latest version for {major_version} major")
+        return resolved_version
+
+    def install_plugin(self, plugin_path: str) -> None:
+        util.run_or_raise_error(
+            ["bin/logstash-plugin", "install", plugin_path],
+            f"Failed to install {plugin_path}")
+
+    def build_logstash(self):
+        print(f"Building Logstash.")
+        util.run_or_raise_error(
+            ["./gradlew", "clean", "bootstrap", "assemble", "installDefaultGems"],
+            "Failed to build Logstash")
+        print(f"Logstash has successfully built.")
+
+    def apply_config(self, config: dict) -> None:
+        with open(os.getcwd() + "/config/pipelines.yml", 'w') as pipelines_file:
```

Comment:
nitpick: we need to be careful not to commit the test config/pipelines.yml, or use `--path.settings` to point the config files to other directory

Reply:
With [the recent commit](https://github.com/elastic/logstash/pull/16467/commits/d26c0484b6681f5720fdb2732996506233bd33a7), it uses `--path.settings`.